modules/text_processor.py

```diff
@@ -126,7 +127,8 @@ def get_data_from_message(self, message):
         date = date_string_to_date(date) if date and date_is_valid(date) else None
         return (keyword, child_name, date)
 
-    def write_to_database(self, message):
+    def write_to_database(self, message_date_tuple):
```

Comment:
Rather than doing a confusing tuple, can you just pass `write_to_database(self, message, received_at)`?